src/loaders/api/register.js

```diff
@@ -40,15 +40,22 @@ export function setupRegisterAPI (agent) {
  * @param {string} [target.licenseKey] the license key of the target to report data to
  * @param {string} target.id the entity ID of the target to report data to
  * @param {string} target.name the entity name of the target to report data to
+ * @param {string} [target.parentId] the ID of the parent target to report data to
  * @returns {RegisterAPI} the api object to be returned from the register api method
  */
 export function buildRegisterApi (agentRef, target) {
   const attrs = {}
   warn(54, 'newrelic.register')
 
   target ||= {}
+  target.eventSource = 'MicroFrontendBrowserAgent'
   target.licenseKey ||= agentRef.info.licenseKey // will inherit the license key from the container agent if not provided for brevity. A future state may dictate that we need different license keys to do different things.
   target.blocked = false
+  if (!target.parentId) {
+    Object.defineProperty(target, 'parentId', {
+      get: () => agentRef.runtime.appMetadata.agents?.[0]?.entityGuid // this is async so must be a getter
```

Comment:
I'm a bit confused as to how this supposed to work. If this method is called prior to RUM response, it's going to be undefined anyways, which means getV2 backfills with the container entity. That means you don't need to set a getter here because `target.parentId` is naturally undefined if it doesn't exist.
And if it's called after... well setting it has no behavioral difference. And if it's called by another MFE, then of course parentId is already provided.
So in summary, this is unnecessary?